core/src/main/scala/org/http4s/headers/X-B3-Sampled.scala

```diff
@@ -17,12 +17,14 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
 import org.http4s.util.Writer
+import cats.parse.{Parser, Parser1, Rfc5234}
 
 object `X-B3-Sampled` extends HeaderKey.Internal[`X-B3-Sampled`] with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[`X-B3-Sampled`] =
-    HttpHeaderParser.X_B3_SAMPLED(s)
+    ParseResult.fromParser(parser, "X-B3-Sampled header")(s)
+  val parser: Parser1[`X-B3-Sampled`] =
```

Comment:
All of our parsers have been `private[http4s]` so far.  Im not committed to that, but let's keep it consistent until we decide.